pointer-events/src/lib.rs

```diff
@@ -18,6 +18,80 @@
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
 #![cfg_attr(all(not(feature = "std"), not(test)), no_std)]
 
+// TODO: We could use the `dpi` crate to distinguish between logical and physical positions
+// and sizes, but it isn't yet `no_std` compatible.
+
+mod buttons;
+
+pub use buttons::{PointerButton, PointerButtons};
+
+/// A unique identifier for the pointing device.
+// TODO: Is this actually available on all platforms, or should
+// we do what browsers do and generate something on our own?
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct PointerId(i64);
+
+// TODO: `f64` seems rather extreme. Can it be smaller?
+// TODO: Would be nice to use `dpi::LogicalSize`, but that is not `no_std`
+/// The size of the bounding box of an input, usually touch.
+///
+/// If this is not provided by the underlying API, platform, or device,
+/// then it will default to a single pixel.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct ContactGeometry {
+    /// The width of the contact geometry.
+    pub width: f64,
+    /// The height of the contact geometry.
+    pub height: f64,
+}
+
+impl Default for ContactGeometry {
+    fn default() -> Self {
+        Self {
+            width: 1.0,
+            height: 1.0,
+        }
+    }
+}
+
+#[non_exhaustive]
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+#[expect(missing_docs, reason = "fill in later")]
+pub enum PointerEventType {
+    PointerOver,
+    PointerEnter,
+    PointerDown,
+    PointerMove,
+    PointerRawUpdate,
+    PointerUp,
+    PointerCancel,
+    PointerOut,
+    PointerLeave,
+    GotPointerCapture,
+    LostPointerCapture,
+    /// TODO: This is from UI Events and may not belong here.
+    AuxClick,
+    /// TODO: This is from UI Events and may not belong here.
+    Click,
+    /// TODO: This is from UI Events and may not belong here.
+    ContextMenu,
+}
+
+/// The type of device that has generated a pointer event.
+// TODO: Instead of non_exhaustive, could have an `Other(String)` variant.
+// This would correspond to the vendor-prefixed name discussed in the
+// pointer events specification.
```

Comment:
I'd prefer not to have that kind of thing, since then you quickly get into a situation with `PointerType::Other("Mouse")` being represent-able, but clearly wrong. Besides, there's value in `Copy` types.

Reply:
I highly prefer the approach taken by `keyboard-types` to add vendor-specific variants:
https://github.com/pyfisch/keyboard-types/blob/b82feeb5ad481f691397da13372cedd178315f41/src/code.rs#L429-L470

Reply:
That just requires a new published version, but shouldn't be a big issue, especially since this is non-exhaustive.